### PR TITLE
[FLINK-6130] [yarn] Fix Consider calling resourceManager#getTerminati…

### PR DESCRIPTION
**Pull request by zhangminglei**
…onFuture() with lock held.

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by zhangminglei** [comment]
@tedyu @zentol It seems there is a problem with travis-ci test if no output within 5 mins then it will print the stacktrace and then kill the watchdog , please help to review and checkout how to solve, I am appreciate it. 
```
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.213 sec - in org.apache.flink.runtime.minicluster.MiniClusterITCase
==============================================================================
Maven produced no output for 300 seconds.
==============================================================================
==============================================================================
The following Java processes are running (JPS)
==============================================================================
1552 Launcher
12642 Jps
8045 surefirebooter5889190340308801783.jar
...
Trying to KILL watchdog (1547).
MVN exited with EXIT CODE: 143.
java.io.FileNotFoundException: build-target/lib/flink-dist*.jar (No such file or directory)
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.<init>(ZipFile.java:220)
	at java.util.zip.ZipFile.<init>(ZipFile.java:150)
	at java.util.zip.ZipFile.<init>(ZipFile.java:121)
	at sun.tools.jar.Main.list(Main.java:1060)
	at sun.tools.jar.Main.run(Main.java:291)
	at sun.tools.jar.Main.main(Main.java:1233)
find: `./flink-yarn-tests/target/flink-yarn-tests*': No such file or directory
PRODUCED build artifacts.
```

**Comment by zhangminglei** [comment]
@tzulitai Hi, Could you please review my code ? I am appreciate it.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java [hunk 1]
```diff
@@ -158,14 +160,16 @@ protected int runApplicationMaster(Configuration config) {
...
 				// ---- (5) start the web monitor
 				// TODO: add web monitor
 			}
+			Object object = future.value().get();
 
-			// wait for resource manager to finish
-			resourceManager.getTerminationFuture().get();
 			// everything started, we can wait until all is done or the process is killed
-			LOG.info("YARN Application Master finished");
+			LOG.info("YARN Application Master finished and the result is " + object.toString());
```
Comment (tzulitai) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java]: I would use a log message formatter here:
`LOG.info("YARN Application Master finished and the result is {}", object);`

**Comment by tzulitai** [comment]
I don't think the timeout was related. We'll see from this Travis run :)

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java (hunk 1 above)
Comment (zhangminglei) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java]: Thanks for review. I will update the code. Very appreciate it.

Comment (tzulitai) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java]: Just another nitpick: the `object` variable could probably use a nicer / more meaningful name :)

**Comment by zhangminglei** [comment]
Travis sucks some time.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java (hunk 1 above)
Comment (zhangminglei) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkApplicationMasterRunner.java]: Very nice.

**Comment by tillrohrmann** [comment]
What is the problem we're trying here to solve?

**Comment by zhangminglei** [comment]
@StephanEwen Hi, How do you think of this refine ?

**Comment by zhangminglei** [comment]
@zentol Please helps review. I am very appreciate it.

**Comment by tillrohrmann** [comment]
@zhangminglei please refrain from pulling more and more people into the loop by directly referencing them given that this is a trivial change and that 5 people are already involved.

Looking at the `YarnFlinkApplicationMasterRunner` I think it could benefit from some more refactoring. For example, I don't think that we need to have an `AbstractYarnFlinkApplicationMasterRunner` base class. Both can be combined in one class. Then we can instantiate the services and runtime components in the constructor of `YarnFlinkApplicationMasterRunner`. That way we can get rid of the lock in the `run` method.

Given that, I'm not sure whether this PR makes then sense anymore. I think we can close it.

**Comment by zhangminglei** [comment]
@tillrohrmann That is right. We can close this PR surely. BTW, Should we make another jira refactoring this class and ```AbstractYarnFlinkApplicationMasterRunner ``` ? Because I watch the ```AbstractYarnFlinkApplicationMasterRunner ``` and find only one subclass. If we do create another jira, You are the reporter and assign it to me. How do you think of this ?

**Comment by tillrohrmann** [comment]
Yes we should create another JIRA issue. You can create it. You can also close FLINK-6130 then.

**Pull request closed without merging**